crypto/hd/algorithm.go

```diff
@@ -1,96 +1,105 @@
 	ethcrypto "github.com/ethereum/go-ethereum/crypto"
 
-	tmcrypto "github.com/tendermint/tendermint/crypto"
-
-	"github.com/cosmos/cosmos-sdk/crypto/keys"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
 
 	"github.com/cosmos/ethermint/crypto/ethsecp256k1"
 )
 
 const (
-	// EthSecp256k1 defines the ECDSA secp256k1 used on Ethereum
-	EthSecp256k1 = keys.SigningAlgo(ethsecp256k1.KeyType)
+	// EthSecp256k1Type defines the ECDSA secp256k1 used on Ethereum
+	EthSecp256k1Type = hd.PubKeyType(ethsecp256k1.KeyType)
 )
 
-// SupportedAlgorithms defines the list of signing algorithms used on Ethermint:
-//  - eth_secp256k1 (Ethereum)
-//  - secp256k1 (Tendermint)
-var SupportedAlgorithms = []keys.SigningAlgo{EthSecp256k1, keys.Secp256k1}
-
-// EthSecp256k1Options defines a keys options for the ethereum Secp256k1 curve.
-func EthSecp256k1Options() []keys.KeybaseOption {
-	return []keys.KeybaseOption{
-		keys.WithKeygenFunc(EthermintKeygenFunc),
-		keys.WithDeriveFunc(DeriveKey),
-		keys.WithSupportedAlgos(SupportedAlgorithms),
-		keys.WithSupportedAlgosLedger(SupportedAlgorithms),
-	}
-}
+var (
+	// SupportedAlgorithms defines the list of signing algorithms used on Ethermint:
+	//  - eth_secp256k1 (Ethereum)
+	//  - secp256k1 (Tendermint)
+	SupportedAlgorithms = keyring.SigningAlgoList{EthSecp256k1, hd.Secp256k1}
+	// SupportedAlgorithmsLedger defines the list of signing algorithms used on Ethermint for the Ledger device:
+	//  - eth_secp256k1 (Ethereum)
+	//  - secp256k1 (Tendermint)
+	SupportedAlgorithmsLedger = keyring.SigningAlgoList{EthSecp256k1, hd.Secp256k1}
+)
 
-func DeriveKey(mnemonic, bip39Passphrase, hdPath string, algo keys.SigningAlgo) ([]byte, error) {
-	switch algo {
-	case keys.Secp256k1:
-		return keys.StdDeriveKey(mnemonic, bip39Passphrase, hdPath, algo)
-	case EthSecp256k1:
-		return DeriveSecp256k1(mnemonic, bip39Passphrase, hdPath)
-	default:
-		return nil, errors.Wrap(keys.ErrUnsupportedSigningAlgo, string(algo))
+// EthSecp256k1Option defines a function keys options for the ethereum Secp256k1 curve.
+// It supports eth_secp256k1 and secp256k1 keys for accounts.
+func EthSecp256k1Option() keyring.Option {
+	return func(options *keyring.Options) {
+		options.SupportedAlgos = SupportedAlgorithms
+		options.SupportedAlgosLedger = SupportedAlgorithmsLedger
 	}
+
 }
 
-// EthermintKeygenFunc is the key generation function to generate secp256k1 ToECDSA
-// from ethereum.
-func EthermintKeygenFunc(bz []byte, algo keys.SigningAlgo) (tmcrypto.PrivKey, error) {
-	if algo != EthSecp256k1 {
-		return nil, fmt.Errorf("signing algorithm must be %s, got %s", EthSecp256k1, algo)
-	}
+var (
+	_ keyring.SignatureAlgo = EthSecp256k1
 
-	return ethsecp256k1.PrivKey(bz), nil
+	// EthSecp256k1 uses the Bitcoin secp256k1 ECDSA parameters.
+	EthSecp256k1 = ethSecp256k1Algo{}
+)
+
+type ethSecp256k1Algo struct {
```

Comment:
empty struct. is this meant to be a wrapper?

Reply:
yup, same as the SDK